src/@types/EventHandler.d.ts

```diff
@@ -0,0 +1,24 @@
+// IDK why this line is necessary, but if I remove it, I can't use EventHandler anywhere
+import type { Awaitable } from 'discord.js';
+
+declare global {
+	/**
+	 * Defines an event handler.
+	 */
+	interface EventHandler {
+		/**
+		 * The name of the event this handler is for
+		 */
+		name: string;
```

Comment:
Could we make this `keyof ClientEvents`? The autocomplete buff would be nice for writing new event handlers, even if the `execute` method still uses `any` params for now

Reply:
I tried this out and it mostly worked - however, it did break the test case where I give the index fake event handlers to make sure it's registering them properly. I'm trying to figure out a workaround.